extensions/BMO/lib/Reports/Attention.pm

```diff
@@ -48,24 +48,37 @@ sub get_bug_list {
 
 sub format_bug_list {
   my ($bugs, $user) = @_;
-
+  my $global_seen  = Bugzilla->request_cache->{attention}->{global_seen};
   my $datetime_now = DateTime->now(time_zone => $user->timezone);
 
   my @formatted_bugs;
+  my %local_seen;
   foreach my $row (@{$bugs}) {
+
+    # Skip if we have already seen this bug locally or globally
+    my $bug_id = $row->{bug_id};
+    next if $local_seen{$bug_id};
+    next if $global_seen->{$bug_id};
+
     my $bug = {
-      id          => $row->{bug_id},
+      id          => $bug_id,
       status      => $row->{bug_status},
       priority    => $row->{priority},
       severity    => $row->{bug_severity},
       summary     => $row->{short_desc},
       changeddate => $row->{delta_ts},
     };
+
     my $datetime = datetime_from($bug->{changeddate});
     $datetime->set_time_zone($user->timezone);
     $bug->{changeddate}       = $datetime->strftime('%Y-%m-%d %T %Z');
     $bug->{changeddate_epoch} = $datetime->epoch;
     $bug->{changeddate_fancy} = time_ago($datetime, $datetime_now);
+
+    # We only want to see a bug id once per page load so mark them seen
+    $local_seen{$bug_id} = 1;
+    $global_seen->{$bug_id} = 1;
```

Comment:
Since we always set both `local_seen` and `global_seen` at the same time, and then check both caches at the start of the loop over the bugs at the same time, isn't `local_seen` unnecessary? When is `local_seen` ever used in a condition where `global_seen` isn't used?

Reply:
Valid point. They are actually performing the same task now that you point it out. I will remove the local_seen part.